src/darwin/Framework/CHIPTests/TestHelpers/MTRTestDeclarations.h

```diff
@@ -55,6 +58,7 @@ NS_ASSUME_NONNULL_BEGIN
 #ifdef DEBUG
 @interface MTRDeviceController (TestDebug)
 - (NSDictionary<NSNumber *, NSNumber *> *)unitTestGetDeviceAttributeCounts;
+- (NSUInteger)unitTestDelegateCount;
```

Comment:
It would be great if we could move away from adding more stuff to this header, and instead use `*_Test.h` headers that are part of the public / test / internal header stack for group of .mm / .h files. The main advantage of that approach is that the test declarations are actually visible to the compiler (and other developers) when compiling the implementation.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/35496